components/TimeSeries.vue

```diff
@@ -0,0 +1,353 @@
+    return Math.max(minTotalAccordionHeight, (Object.keys(appStore.timeSeriesData!).length * minAccordionHeight));
+  } else {
+    return minTotalAccordionHeight;
+  }
+});
+// Share available height equally between open accordions
+const containerHeightPx = computed(() => maxTotalAccordionHeight.value / props.openedAccordions.length);
+const containerId = computed(() => `${props.seriesId}-container`);
+// Assign an x-position to y-values. Nth value corresponds to "N+1th day" of simulation.
+const data = computed(() => {
+  return appStore.timeSeriesData![props.seriesId].map((value, index) => [index + 1, value]);
+});
+const seriesMetadata = computed((): DisplayInfo | undefined => {
+  return appStore.metadata?.results?.time_series.find(({ id }) => id === props.seriesId);
+});
+const usePlotLines = props.seriesId === "hospitalised"; // https://mrc-ide.myjetbrains.com/youtrack/issue/JIDEA-118/
+const usePlotBands = props.seriesId === "hospitalised" || props.seriesId === "infect"; // https://mrc-ide.myjetbrains.com/youtrack/issue/JIDEA-118/
+// On zooming, the y-axis automatically rescales to the data (ignoring the plotLines). We want the plotLines
+// to remain visible, so we limit the y-axis' ability to rescale, by defining a minimum range.
+const minRange = computed(() => {
+  if (usePlotLines) {
+    return appStore.capacitiesData?.reduce((acc, { value }) => Math.max(acc, value), 0);
+  } else {
+    return undefined;
+  }
+});
+const capacitiesPlotLines = computed(() => {
+  const lines = Array<Highcharts.AxisPlotLinesOptions>();
+
+  if (!usePlotLines) {
+    return lines;
+  }
+
+  appStore.capacitiesData?.forEach(({ id, value }) => {
+    const capacityLabel = appStore.metadata?.results.capacities.find(({ id: capacityId }) => capacityId === id)?.label;
+
+    lines.push({
+      color: plotLinesColor,
+      label: {
+        text: `${capacityLabel}: ${value}`,
+        style: {
+          color: plotLinesColor,
+        },
+      },
+      width: 2,
+      value,
+      zIndex: 4, // Render plot line and label in front of the series line
+    });
+  });
+
+  return lines;
+});
+
+const interventionsPlotBands = computed(() => {
+  const bands = Array<Highcharts.AxisPlotBandsOptions>();
+
+  if (!usePlotBands) {
+    return bands;
+  }
+
+  appStore.interventionsData?.forEach(({ id, start, end }) => {
+    const interventionLabel = appStore.metadata?.results.interventions.find(({ id: interventionId }) => interventionId === id)?.label;
+    const text = `${interventionLabel ? `${interventionLabel} : ` : ""}Days ${start} to ${end}`;
+    bands.push({ from: start, to: end, color: plotBandsColor, label: { text } });
+  });
+
+  return bands;
+});
+
+const allCharts = computed(() => {
+  // Excludes undefineds. When a chart is destroyed, Highcharts.charts retains its index, but the value is undefined.
+  return Array.from(Highcharts.charts).filter(chart => chart) as (Highcharts.Chart)[];
+});
```

Comment:
So this gets all charts loaded anywhere in the page? Again, this component doesn't feel like quite the right place for any code that's dealing with all charts. Could this go in a `TimeSeriesList` component?
Also, this might be a bit dangerous if we potentially end up using Highcharts for other elements in the page. Could we get to all charts by id or ref?

Reply:
Good point on 'dangerous'! Now we store the charts as a dict belonging to the TImeSeriesList component.